cellpack/autopack/writers/__init__.py

```diff
@@ -7,10 +7,99 @@
 import os
 import numpy
 from collections import OrderedDict
+import typing
 
 from cellpack import autopack
+from mdutils.mdutils import MdUtils
+import pandas as pd
+from pathlib import Path
 
 
+class MarkdownWriter(object): 
+
+    def __init__(self, title:str, output_path:Path, report_name:str):
+        self.title = title
+        self.output_path = output_path
+        self.report_md = MdUtils(
+            file_name=str(self.output_path / report_name ),
+            title=title,
+        )
+
+    # level is the header style, can only be 1 or 2
+    def add_header(self, header:str, level:int=2):
+        self.report_md.new_header(
+            level = level,
+            title = header,
+            add_table_of_contents="n"
+        )
+
+    def add_table(self, header:str, table:pd.DataFrame, text_align="center"):
+        self.report_md.new_header(
+            level=1,
+            title=header,
+            add_table_of_contents="n",
+        )
+        
+        text_list = []
+        for row in table.values.tolist():
+            for item in row:
+                text_list.append(item)
+
+        self.report_md.new_table(
+            columns=table.shape[1],
+            rows=table.shape[0],
+            text=text_list,
+            text_align=text_align
+        )
+
+    def add_table_from_csv(self, header:str, filepath:Path, text_align="center"):
+        self.report_md.new_header(
+            level=1,
+            title=header,
+            add_table_of_contents="n",
+        )
+        
+        table = pd.read_csv(filepath)
+
+        text_list = []
+        for row in table.values.tolist():
+            for item in row:
+                text_list.append(item)
+
+        self.report_md.new_table(
+            columns=table.shape[1],
+            rows=table.shape[0],
+            text=self.text_list,
+            text_align=text_align
+        )
+    def write_file(self):
+        self.report_md.create_md_file()
+
+    # Image text must be a list, if list is not same length as list of filepaths, only 1st item in image_text is used
+    def add_images(self, header:str, image_text:typing.List[str], filepaths:typing.List[str]):
+        self.report_md.new_header(
+            level=1,
+            title=header,
+            add_table_of_contents="n",
+        )
+        if len(image_text) == len(filepaths):
+            for i in range(len(filepaths)):
+                self.report_md.new_line(
+                    self.report_md.new_inline_image(
+                        text=image_text[i],
+                        path=str(self.output_path / filepaths[i])
```

Comment:
We want this to be `output_image_location` and not `output_path`. The distinction is as follows:
1. `output_image_location` refers to the path used in the report to link to the image. This can be the AWS bucket path after the github workflow runs.
2. `output_path` is the path to which the markdown file itself is written. This is typically the root directory where packing outputs are stored.